### PR TITLE
docs(SelectPanel): update docs with keyboard interactions

### PR DESCRIPTION
**Pull request by joshblack**
This PR updates the docs for `SelectPanel` to document the keyboard behaviors supported by the component. It also adds in an initial Status section and initial Props.

**Questions**

- Is the interaction guidance here appropriate? I wasn't sure if keyboard interaction docs should live here or somewhere else 👀 
- What's the best way to write out the props for SelectPanel using the existing MDX components?
- Is the ComponentChecklist correct?

**References**

- https://github.com/github/primer/issues/1021

### CODE REVIEW

File: docs/content/SelectPanel.mdx [hunk 1]
```diff
@@ -66,4 +66,60 @@ function DemoComponent() {
...
+
+A user may use `Tab` and `Shift+Tab` to navigate between the filter input, list of
+items, action buttons, and the close button.
+
+When focus is on the list of items, pressing `ArrowDown` and `ArrowUp` will allow navigation between items. Support for `Home` and `End` is also provided to quickly navigate to the first and last items in the list, respectively. Pressing `Space` will select an item.
+
+The `Enter` key may be used when focus is within the `SelectPanel` as an
+alternative to activating the save button.
+
 ## Component props
```
Comment (colebemis) [inline docs/content/SelectPanel.mdx:84]: While we're touching this file, let's rename this to just "Props" for consistency.
```suggestion
## Props
```

File: docs/content/SelectPanel.mdx [hunk 2]
```diff
@@ -66,4 +66,60 @@ function DemoComponent() {
 render(<DemoComponent />)
 ```
 
+## Interactions
```
Comment (colebemis) [inline docs/content/SelectPanel.mdx:69]: Could we call this section "Accessibility" to match the PageLayout docs: https://primer.style/react/PageLayout#accessibility
```suggestion
## Accessibility
```

File: docs/content/SelectPanel.mdx [hunk 3]
```diff
@@ -66,4 +66,60 @@ function DemoComponent() {
...
+items, action buttons, and the close button.
+
+When focus is on the list of items, pressing `ArrowDown` and `ArrowUp` will allow navigation between items. Support for `Home` and `End` is also provided to quickly navigate to the first and last items in the list, respectively. Pressing `Space` will select an item.
+
+The `Enter` key may be used when focus is within the `SelectPanel` as an
+alternative to activating the save button.
+
 ## Component props
+
+<PropsTable>
```
Comment (colebemis) [inline docs/content/SelectPanel.mdx:86]: We usually have a heading for the name of the component above the props table
```suggestion

### SelectPanel

<PropsTable>
```

File: docs/content/SelectPanel.mdx [hunk 4]
```diff
@@ -66,4 +66,60 @@ function DemoComponent() {
...
+    type={`(
+  open: boolean,
+  gesture:
+    | 'anchor-click'
+    | 'anchor-key-press'
+    | 'click-outside'
+    | 'escape'
+    | 'selection'
+) => void`}
+    required={true}
```
Comment (colebemis) [inline docs/content/SelectPanel.mdx:98]:
```suggestion
    required
```

File: docs/content/SelectPanel.mdx [hunk 5]
```diff
@@ -66,4 +66,60 @@ function DemoComponent() {
...
+  />
+  <PropsTableRow name="title" type="string" />
+  <PropsTableRow name="inputLabel" type="string" />
+  <PropsTableRow name="inputPlaceholder" type="string" />
+  <PropsTableRow name="overlayProps" type="Partial<OverlayProps>" />
+</PropsTable>
```
[the remark this answers is not in the document]
Comment (joshblack) [reply docs/content/SelectPanel.mdx:106]: @colebemis I was waiting to hear back over in https://github.slack.com/archives/C02NUUQ9C30/p1663172773252429 to see what the recommendations are for the union in the type 👀

File: docs/content/SelectPanel.mdx [hunk 6]
```diff
@@ -65,4 +65,89 @@ function DemoComponent() {
...
+  />
+  <PropsTableRow
+    name="overlayProps"
+    type="Partial<OverlayProps>"
+    description={
+      <>
+        See <a href="/react/Overlay#props">Overlay props</a>.
+      </>
+    }
+  />
```
Comment (colebemis) [inline docs/content/SelectPanel.mdx:121]: This looks like a duplicate

**Review by colebemis — APPROVED** [review]

**Pull request merged**